detect/region.go

```diff
@@ -0,0 +1,62 @@
+package detect
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/urfave/cli"
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/cli/utils"
+
+	"github.com/Scalingo/go-scalingo/v4/debug"
+)
+
+// GetRegionFromGitRemote returns the region name extracted from remotes URL of the current Git repository
+// If not found it returns an empty string
+func GetRegionFromGitRemote(c *cli.Context, rc *config.RegionsCache) string {
+	remoteName := c.GlobalString("remote")
+
+	if dir, ok := utils.DetectGit(); ok {
+		remotes, err := utils.ScalingoGitRemotes(dir)
+		if err != nil {
+			debug.Println(err)
```

Comment:
Why are the errors not returned and directly printed in a debug log? This is not the choice made for apps (cf. `GetAppNameFromGitRemote`).

Reply:
It's the same for the app part (`CurrentApp`), at first, this error was simply ignored, @johnsudaar thought it could be convenient to print it in a debug log.  It's not a blocking error, we do not want to make anything out of it. We simply return an empty string and the `--region` flag is use instead.